open-api/lang-tools/src/main/java/io/deephaven/lang/completion/ChunkerCompleter.java

```diff
@@ -1026,8 +1033,8 @@ private Optional<Class<?>> resolveScopeType(
             // This is pretty hacky, but our only use case here is looking for table loads, so...
             if (assignment.getValue() instanceof IsScope) {
                 final List<IsScope> subscope = ((IsScope) assignment.getValue()).asScopeList();
-                if (!subscope.equals(scope)) {
-                    result = resolveScopeType(request, subscope);
+                if (!alreadyVisited.contains(subscope)) {
```

Comment:
could probably just do `if (alreadyVisited.add(subscope))`, but not really important.